generator/lib/src/field_element_ex.dart

```diff
@@ -133,14 +133,16 @@ extension FieldElementEx on FieldElement {
       }
 
       // Validate indexes
-      if ((primaryKey != null || indexed != null) &&
-          (![RealmPropertyType.string, RealmPropertyType.int, RealmPropertyType.objectid, RealmPropertyType.uuid].contains(type.realmType) ||
-              type.isRealmCollection)) {
+      if ((primaryKey != null || indexed != null) && !(type.realmType?.mapping.indexable ?? false)) {
         final file = span!.file;
         final annotation = (primaryKey ?? indexed)!.annotation;
+        final listOfIndexableTypes = RealmPropertyType.values //
+            .map((t) => t.mapping)
+            .where((m) => m.indexable) //
+            .expand((m) => [m.type, m.nullableType]);
```

Comment:
This will likely make the error message fairly long - do we need to explicitly list the nullable types? Can't we change the error message to something like `Realm only supports indexes on fields of type int, bool, ... as well as their nullable versions`?

Reply:
Fixed